python/imageutils.py

```diff
@@ -0,0 +1,481 @@
+
+                            seek_times.append(seek_end)
+                            write_times.append(write_end)
+
+                            if seek_end > max_seek:
+                                max_seek = seek_end
+                            if write_end > max_write:
+                                max_write = write_end
+                            if seek_end < min_seek:
+                                min_seek = seek_end
+                            if write_end < max_write:
+                                min_write = write_end
+
+
+                    #print time()-t
+                    
+                    #avg_seek = avg_seek / (shape_x * shape_z)
+                    #avg_write = avg_write / (shape_x * shape_z)
+                    
+                    #print "Maximum seek time: {0}\t Maximum write time: {1}".format(max_seek, max_write)
+                    #print "Minimum seek time: {0}\t Minimum write time: {1}".format(min_seek, min_write)
+                    #print "Average seek time: {0}\t Average write time: {1}".format(avg_seek, avg_write)
+                    #print "Std seek: {0}\t Std write: {1}".format(np.std(seek_times), np.std(write_times))
+
+                    #if end_x == -1:
+                    #    end_x = 0
+                    #if end_z == -1:
+                    #    end_z = 0
+                    #prev_b_end = (block_y + shape[0], block_z + end_z, block_x + end_x)
+                    #print "End position of write: {0}".format(prev_b_end)
+    
+    #will only work if the amount of blocks that fit in memory represents a rectangular prism
+    def cp_block_slice(self, legend, mem):
+        with open(self.filepath, "r+b") as reconstructed:
+
+            rec_dims = self.proxy.header.get_data_shape()
+
+            y_size = rec_dims[0]
+            z_size = rec_dims[1]
+            x_size = rec_dims[2]
+
+            with open(legend, "r") as legend:
+
+                remaining_mem = mem
+                data = None
+                for block_name in legend:
+                    
+                    block_name = block_name.strip()
+                    print "Reading block {0}".format(block_name)
+
+                    block_pos = block_name[:-4].split('_')
+                    start_x = 0
+                    start_z = 0
+                    start_y = 0
+
+                    #we will start our (partial) slice at these block coordinates
+                    if remaining_mem == mem:
+                        start_x = int(block_pos[-1].split('.')[0])
+                        start_z = int(block_pos[-2])
+                        start_y = int(block_pos[-3])
+                        
+                    block = nib.load(block_name)
+                    block_shape = block.header.get_data_shape()
+                    bytes_per_voxel = self.proxy.header['bitpix']/8
+                    block_bytes = 352 + bytes_per_voxel*(block_shape[0]*block_shape[1]*block_shape[2]) + 4
```

Comment:
i found a class variable in nibabel (single_vox_offset) that i could use to represent the nifti-1 (348bytes) and nifti-2 (540bytes) header size. The problem is that this method could work with minc blocks too (all that's really important is that we're writing to a nifti image), but I haven't figured out how many bytes are loaded into memory for the minc header yet.